igcollect/sfp_dom.py

```diff
@@ -235,51 +306,161 @@ class IFReq(ctypes.Structure):
     ]
 
 
-class SFPData:
-    FN_MAP = {
-        'temperature': a2_offset_to_celsius,
-        'bias_current_lane1': a2_offset_to_milliamps,
-        'sfp_voltage': a2_offset_to_volts,
-        'tx_power_lane1': a2_offset_to_milliwatts,
-        'rx_power_lane1': a2_offset_to_milliwatts,
+class SFFChannelData:
+    bias_current = None
```

Comment:
maybe you want to use dataclasses here, or old-school `@property`. then you don't need the `values` getter and setter

Reply:
I gave dataclasses a try. 

The biggest issue is because I pre-create the `SFFChannelData` objects when I create the `SFFData` object, but I only set the values later, when `SFFData` calls the `_parse_eeprom_data()` function. 

I also rely later on the data from `SFFChannelData` already in a dict, which is provided by the `get_values()` function.